app/src/main/java/com/example/atheneum/fragments/newRequest.java

```diff
@@ -99,12 +159,128 @@ public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+        MenuItemCompat.setShowAsAction(item, MenuItemCompat.SHOW_AS_ACTION_COLLAPSE_ACTION_VIEW | MenuItemCompat.SHOW_AS_ACTION_IF_ROOM);
+        MenuItemCompat.setActionView(item, sv);
+        sv.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(final String query) {
+                Log.d(TAG, "On Query Text Submit was Called");
+                db = FirebaseDatabase.getInstance();
+                ref = db.getReference("books");
+
+
+                ref.orderByChild("books").addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                        searchAvailableBook.clear();
+                        for (DataSnapshot child : dataSnapshot.getChildren()) {
+                            Book book = child.getValue(Book.class);
+                            if(searchCheck(book, query)){
+                                searchAvailableBook.add(book);
+                            }
+
+                        }
+                        if (searchAvailableBook.isEmpty()) {
+                            Toast.makeText(getActivity(), "No exact matches found for search query", Toast.LENGTH_SHORT).show();
+                        }
+                        availableAdapter = new requestAdapter(newRequest.this.context, R.layout.request_list_item, searchAvailableBook);
+                        availableBookList.setAdapter(availableAdapter);
+                    }
+
+                    @Override
+                    public void onCancelled(@NonNull DatabaseError databaseError) {
+                        Log.d(TAG1, "On Cancelled of Options Menu");
+                    }
                 });
+                return false;
+            }
 
+            @Override
+            public boolean onQueryTextChange(String s) {
+                return false;
+            }
+        });
+        sv.setIconifiedByDefault(false);
+        sv.setOnSearchClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG1, "On CLICKC was Called");
             }
         });
 
-        return this.view;
+        super.onCreateOptionsMenu(menu,inflater);
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        Log.d(TAG, "OUTER ONQueryTextSubmit Called");
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String newText) {
+        Log.d(TAG, "onQueryTextChange"+newText);
+        return false;
+    }
+
+    /**
+     * search an availabel book by title/author/description
+     * @param book
+     * @param query
+     * @return
+     */
+    public boolean searchCheck(Book book, String query){
+        Log.d(TAG1, "Get Query "+query+query.length());
```

Comment:
This doesn't satisfy the requirement:

US 03.01.01
As a borrower, I want to specify a set of keywords, and search for all books that are not currently accepted or borrowed whose description contains all the keywords.

You have to match based on the book description not on the author/title. An easy way to check for a list of keywords would be to to have your search string be a space separated list of keywords ex. "deep work focus" and then you can filter a list of books based on books whose descriptions contain all those keywords.